src/elf/section_header.rs

```diff
@@ -442,13 +442,19 @@ if_alloc! {
             self.sh_addr as usize..(self.sh_addr as usize).saturating_add(self.sh_size as usize)
         }
         /// Parse `count` section headers from `bytes` at `offset`, using the given `ctx`
+        /// Assuming this is read from the whole file, it will check offset.
         #[cfg(feature = "endian_fd")]
         pub fn parse(bytes: &[u8], mut offset: usize, mut count: usize, ctx: Ctx) -> error::Result<Vec<SectionHeader>> {
-            use scroll::Pread;
             // Zero offset means no section headers, not even the null section header.
             if offset == 0 {
                 return Ok(Vec::new());
             }
+            Self::parse_raw(bytes, offset, count, ctx)
+        }
+        /// Parse `count` section headers from `bytes` at `offset`, using the given `ctx`
```

Comment:
update document comment here to explain what this does., and why it's different; I'm also not sure "raw" is the best name here/worried we're getting an api divergence from e.g., `parse_lossy` in the mach-o PR for example

Reply:
I'll adopt `parse_lossy` to follow PR #386

Reply:
Sorry I should have been clearer; so in general, `parse_impl` should be a private fn that is used by e.g., two other (maybe `pub`) function bodies, (really the idiom I use is any privately shared body has `_impl` e.g., `fn something_impl`).

What I meant by the api similarity was the new `pub` function should perhaps be called `parse_lossy`, but looking again at this PR, I think you're right, and my suggestion was wrong (apologies!), in that it isn't parsing in a lossy manner, but rather it's parsing the raw headers *at* or *from* a specific byte slice.

So I think in this case a better name is `parse_from` or `parse_at` (I think I'm partial to `parse_from`?)

Alternatives that I don't like but considered:
1. parse_relative
2. parse_headers_from
3. parse_headers_at

If you change the function back to `parse_from` I think we're ok to merge.

One last thing, I don't really like that the user would have to know that `count == 0` will "turn on" the `empty_sh.sh_size`  condition, but I'm not sure of a better way to factor this, and it was also technically true of the original function anyway, so it's fine i guess :)

Reply:
Honestly, I dont really care about the naming, so if you have a preference over `parse_from`, let's use it :)